service/frontend/workflow_handler.go

```diff
@@ -3711,11 +3711,26 @@ func (wh *WorkflowHandler) StartBatchOperation(
 		Namespace: request.GetNamespace(),
 		Query:     batcher.OpenBatchOperationQuery,
 	})
-	if err != nil {
-		return nil, err
-	}
-	if countResp.GetCount() >= int64(wh.config.MaxConcurrentBatchOperation(request.GetNamespace())) {
-		return nil, serviceerror.NewUnavailable("Max concurrent batch operations is reached")
+	if err == nil {
+		if countResp.GetCount() >= int64(wh.config.MaxConcurrentBatchOperation(request.GetNamespace())) {
+			return nil, serviceerror.NewUnavailable("Max concurrent batch operations is reached")
+		}
+	} else {
+		listResp, err := wh.ListOpenWorkflowExecutions(ctx, &workflowservice.ListOpenWorkflowExecutionsRequest{
+			Namespace: request.GetNamespace(),
```

Comment:
Should we also set `MaximumPageSize` here? we can use either `int64(wh.config.MaxConcurrentBatchOperation(request.GetNamespace()))` or `1`, but I think it's a good idea to put a cap on it here. 

If we do the former, we can also check `openCount` against the `MaxConcurrentBatchOperation` parameter instead of 1 to be consistent. Although, if that number is high, querying db for a list of workflows can become slow.